activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -173,10 +173,13 @@ def initialize(config_or_deprecated_connection, deprecated_logger = nil, depreca
       end
 
       def synchronized=(synchronized) # :nodoc:
```

Comment:
`synchronized` sounds boolean, but we've switched to actually needing its object value. Perhaps we should rename to `lock_thread` for now, while acknowledging that we already ought to rename that on the pool given the execution state changes?